packages/language-service/ivy/test/references_spec.ts

```diff
@@ -0,0 +1,514 @@
+      {name: _('/app.html'), contents: '{{myProp}}'}
+    ]);
+    const result = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    // Angular LS only returns the template ref since TSLS will return the class member
+    expect(result.length).toBe(1);
+    expect(result[0].fileName).toContain('app.html');
+    expect(result[0].textSpan).toEqual('myProp');
+  });
+
+  it('getting references from app component', () => {
+    const appFile = getBasicAppFileContents({
+      classContents: 'myProp: string',
+      componentMeta: {templateUrl: 'app.html'},
+    });
+    const {text, cursor} = extractCursorInfo('{{myP¦rop}}');
+    env = LanguageServiceTestEnvironment.setup([
+      {name: _('/app.ts'), contents: appFile, isRoot: true},
+      {name: _('/app.html'), contents: text},
+    ]);
+    const refs = getReferencesAtPosition(_('/app.html'), cursor)!;
+    // Angular LS returns the template ref _and_ component member because TSLS doesn't know how to
+    // find it from the template.
+    expect(refs.length).toBe(2);
+    assertFileNames(refs, ['app.html', 'app.ts']);
+    assertTextSpans(refs, ['myProp']);
+  });
+
+  it('should work for method calls', () => {
+    const fileWithCursor = getBasicAppFileContents({
+      classContents: 'setTitle(s: string) {}',
+      componentMeta: {template: `<div (click)="set¦Title('title')"></div>`},
+    });
+    const {text, cursor} = extractCursorInfo(fileWithCursor);
+    env = LanguageServiceTestEnvironment.setup([
+      {name: _('/app.ts'), contents: text, isRoot: true},
+    ]);
+    const refs = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    expect(refs.length).toBe(2);
+
+    assertFileNames(refs, ['app.ts']);
+    assertTextSpans(refs, ['setTitle']);
+  });
+
+  it('should work for method call arguments', () => {
+    const fileWithCursor = getBasicAppFileContents({
+      classContents: `
+    title: string;
+    setTitle(s: string) {}`,
+      componentMeta: {template: `<div (click)="setTitle(ti¦tle)"></div>`},
+    });
+    const {text, cursor} = extractCursorInfo(fileWithCursor);
+    env = LanguageServiceTestEnvironment.setup([
+      {name: _('/app.ts'), contents: text, isRoot: true},
+    ]);
+    const refs = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    expect(refs.length).toBe(2);
+
+    assertTextSpans(refs, ['title']);
+  });
+
+  it('should work for property writes', () => {
+    const fileWithCursor = getBasicAppFileContents({
+      classContents: `title: string;`,
+      componentMeta: {template: `<div (click)="ti¦tle = 'newtitle'"></div>`},
+    });
+    const {text, cursor} = extractCursorInfo(fileWithCursor);
+    env = LanguageServiceTestEnvironment.setup([
+      {name: _('/app.ts'), contents: text, isRoot: true},
+    ]);
+    const refs = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    expect(refs.length).toBe(2);
+
+    assertTextSpans(refs, ['title']);
+  });
+
+  describe('references', () => {
+    it('should work', () => {
```

Comment:
```suggestion
    it('should work for element references', () => {
```
Could we add a test for template references?